libs/R-TypeLogic/Global/Components/ActionName.hpp

```diff
@@ -0,0 +1,32 @@
+/*
+** EPITECH PROJECT, 2022
+** Project
+** File description:
+** ActionName
+*/
+
+#ifndef ACTIONNAME_HPP_
+#define ACTIONNAME_HPP_
+
+#include "Component/Component.hpp"
+#include "R-TypeLogic/Global/SharedResources/ButtonActionMap.hpp"
+
+namespace ecs
+{
+    /// @brief Action Name store the key as enum to the corresponding action
+    class ActionName : public Component {
```

Comment:
Maybe you can specify which type of action (like ButtonActionName). It suggests updating filename, ifndef, documentation, ...